core/app/models/spree/stock_item.rb

```diff
@@ -96,12 +96,14 @@ def process_backorders(number)
       end
 
       def conditional_variant_touch
-        # the variant_id changes from nil when a new stock location is added
-        stock_changed = (count_on_hand_changed? && count_on_hand_change.any? { |c| c < 1 }) || variant_id_changed?
+        variant.touch if Spree::Config.inventory_cache_threshold.nil? || count_breaks_threshold?
+      end
 
-        if !Spree::Config.binary_inventory_cache || stock_changed
-          variant.touch
-        end
+      def count_breaks_threshold?
+        # the variant_id changes from nil when a new stock location is added
+        Spree::Config.inventory_cache_threshold &&
+        (count_on_hand_changed? && count_on_hand_change.any? { |c| c < Spree::Config.inventory_cache_threshold }) ||
```

Comment:
so if i wanted to maintain binary inventory cache, I would set this to 1 then? I wonder if there's a way to issue a deprecation warning on the binary_inventory_cache preference and make it set the inventory_cache_threshold response. Maybe a method on Spree::Config?